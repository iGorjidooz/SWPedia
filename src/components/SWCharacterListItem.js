import React from 'react';
import { Link } from 'react-router-dom';

const SWCharacterListItem = ({ character, listIndex }) => {
   return (
      <Link to={`/character/${character.id}`}>
         <h3>{listIndex} {character.name}</h3>
         <p>Birth Year: {character.birth_year}</p>
         <p>Gender: {character.gender}</p>
      </Link>
   );
};

export default SWCharacterListItem;