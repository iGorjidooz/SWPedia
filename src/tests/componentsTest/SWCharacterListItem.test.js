import React from 'react';
import { shallow } from 'enzyme';
import peopleFixtures from '../fixtures/peopleFixtures';
import SWCharacterListItem from '../../components/SWCharacterListItem';

 test('should render SWCharacterListItem correctly', () => {
    const wrapper = shallow(
      <SWCharacterListItem character={peopleFixtures[1]} listIndex={peopleFixtures[1].id} />
    );
    expect(wrapper).toMatchSnapshot();
 })