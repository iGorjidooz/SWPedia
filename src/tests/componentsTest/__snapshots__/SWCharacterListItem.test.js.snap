// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SWCharacterListItem correctly 1`] = `
<Link
  to="/character/2"
>
  <h3>
    2
     
    C-3PO
  </h3>
  <p>
    Birth Year: 
    112BBY
  </p>
  <p>
    Gender: 
    n/a
  </p>
</Link>
`;
